src/fivetran-destination/src/destination/dml.rs

```diff
@@ -388,6 +407,152 @@ async fn update_file(
     Ok(())
 }
 
+async fn delete_files(
+    schema: &str,
+    table: &Table,
+    file_config: &FileConfig,
+    client: &tokio_postgres::Client,
+    delete_files: &[String],
+) -> Result<(), anyhow::Error> {
+    // TODO(parkmycar): Make sure table exists.
+    // TODO(parkmycar): Retry transient errors.
+
+    // Copy into a temporary table, which we then merge into the destination.
+    let ts = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .context("time went backwards!")?;
+    // TODO(parkmycar): Make sure table name is <= 255 characters.
+    let temp_table_name = format!("fivetran_{}_temp_delete_{}", table.name, ts.as_millis());
+
+    let schema_name = escape::escape_identifier(schema);
+    let table_name = escape::escape_identifier(&table.name);
+    let temp_table_name = escape::escape_identifier(&temp_table_name);
+
+    let columns = table
+        .columns
+        .iter()
+        .map(|col| {
+            let mut ty: Cow<'static, str> = utils::to_materialize_type(col.r#type())?.into();
+            if let Some(d) = &col.decimal {
+                ty.to_mut()
+                    .push_str(&format!("({}, {})", d.precision, d.scale));
+            }
+
+            Ok(ColumnMetadata {
+                name: col.name.to_string(),
+                ty,
+                is_primary: col.primary_key || col.name.to_lowercase() == FIVETRAN_SYSTEM_COLUMN_ID,
+            })
+        })
+        .collect::<Result<Vec<_>, anyhow::Error>>()?;
+
+    // Create our temporary table that we'll copy the delete files into.
+    let defs = columns
+        .iter()
+        .map(|col| format!("{} {}", col.name, col.ty))
+        .join(",");
+    let create_table_stmt = format!("CREATE TABLE {schema_name}.{temp_table_name} ({defs})");
+    client.batch_execute(&create_table_stmt).await?;
+
+    // TODO(parkmycar): Ideally we'd use the COPY FROM protocol here, but we need to add support
+    // for compression and encryption.
+    let cols = columns
+        .iter()
+        .enumerate()
+        .map(|(idx, col)| format!("${}::{}", idx + 1, col.ty))
```

Comment:
Why is the cast needed here? It should be implied by the table desc during insert.

Reply:
Removed entirely in favor of `COPY FROM`